pkg/controller/v1beta1/inferenceservice/reconcilers/service/service_reconciler.go

```diff
@@ -64,6 +64,10 @@ func createService(componentMeta metav1.ObjectMeta, componentExt *v1beta1.Compon
 	} else {
 		port, _ = strconv.Atoi(constants.InferenceServiceDefaultHttpPort)
 	}
+	if componentExt.Batcher != nil {
```

Comment:
I guess it is the same problem for logger, could you help test that ? Thanks!

Reply:
Might need to add componentExt.logger != nil as well as the multi model case when agent is injected